acceptance/suites/tests/authorization/use_csr_auth.rb

```diff
@@ -0,0 +1,141 @@
+test_name "(TK-293) TK-AUTH uses certificate extensions for authentication" do
+
+server = master.puppet['certname']
+ssldir = master.puppet['ssldir']
+confdir = master.puppet['confdir']
+
+# Do we have a functioning cert?
+step "Confirm agent can connect with existing cert" do
+  agents.each do |a|
+    if (not_controller(a))
+      rc = on(a,
+              puppet("agent --test --server #{server} --detailed-exitcodes"),
+              {:acceptable_exit_codes => [0,2]})
+    end
+  end
+end
+
+# Not anymore we don't
+step "Revoke and destroy the existing cert on the server" do
```

Comment:
I don't see a revoke happening here.  This is probably just a nit, and cert destruction is probably sufficient for the needs of this test.  Revocation involves appending the certificate to the Certificate Revocation List, and bouncing the server (probably NOT hup).